### ISSUE TITLE
Columns for the song list not editable?

### ISSUE THREAD

**Issue by kaffeeundsalz** [quodlibet/quodlibet#3925]
Not using the bug template for reporting this because I don't know whether this is intended behavior. I was trying to rename a custom column I created for QuodLibet's song list. Clicking the Edit… button under Preferences > Song List opens up the Edit Columns dialog with a list that contains both the standard columns (those which are configurable via the checkboxes in the Visible Columns section) and custom columns.

I can **add** and **remove** entries in this list by using the corresponding buttons on the right-hand side on the window. But I can't figure out a way to **edit** existing entries. Double-clicking as well as right-clicking a row does nothing. This has mainly two disadventages for me: First, if I make a mistake in the tag expression, I can't correct it (I have to remove it and add a new one). Second, I'm stuck with whatever QuodLibet suggests for the column name because the Description is not exposed as an editable value when adding a new tag expression.

Should I be able to edit existing rows in  the Edit Columns dialog or does it work as it's supposed to? Using QuodLibet 4.4.0 under macOS 12.2.1 Monterey.

![ScreenShot 2022-02-28-#000023](https://user-images.githubusercontent.com/4942960/156056277-09a1f48e-30cc-48ef-9e04-32f95284fc75.png)

**Comment by declension** [comment]
> I can add and remove entries in this list by using the corresponding buttons on the right-hand side on the window. But I can't figure out a way to edit existing entries

Yes that's "intentional" behaviour currently, or rather: nobody has written the edit UI for that. It's not too hard I expect, just a bit fiddly. Perhaps you could convert this issue to be an enhancement for that then...

> Second, I'm stuck with whatever QuodLibet suggests for the column name because the Description is not exposed as an editable value when adding a new tag expression.

Ah, that's another point entirely. The descriptions aren't user-editable at all.

**Comment by kaffeeundsalz** [comment]
@declension As always, thank you for putting this straight. I can see that this seems to be related to #3921 where @afontenot also mentions that descriptions are not currently editable:

> Quod Libet currently uses some (flawed) heuristics to pick a column name when a custom tag expression is used for a column; after implementing this feature, Quod Libet would use the description field as the column name instead.

Looking forward to see this improved :)
